core/src/main/java/com/digitalpebble/stormcrawler/protocol/AbstractHttpProtocol.java

```diff
@@ -65,6 +76,134 @@ public void configure(Config conf) {
+                if (!interfaceOK) {
+                    throw new RuntimeException("Class "
+                            + proxyManagerImplementation
+                            + " does not implement ProxyManager");
+                }
+            } catch (ClassNotFoundException e) {
+                throw new RuntimeException("Can't load class "
+                        + proxyManagerImplementation);
+            }
+
+            LOG.info("loaded proxy manager class: {}", proxyManagerClass.getName());
+
+            // create boolean to track if proxy setup succeeded
+            boolean success = true;
+
+            // map proxy type to the proper manager
+            if (proxyFile != null && !proxyFile.isEmpty()) {
+                try {
+                    // create new proxy manager from file
+                    proxyManager = (ProxyManager) proxyManagerClass.newInstance();
+                    proxyManager.configure(proxyRotationScheme, proxyFile);
+                } catch (FileNotFoundException | InstantiationException | IllegalAccessException e) {
+                    LOG.error("failed to create proxy manager `" + proxyFile + "`", e);
+                    success = false;
+                }
+            } else {
+                // load remaining values for single proxy
+                String proxyType = ConfUtils.getString(conf, "http.proxy.type", "HTTP");
+                int proxyPort = ConfUtils.getInt(conf, "http.proxy.port", 8080);
+                String proxyUsername = ConfUtils.getString(conf, "http.proxy.user", null);
+                String proxyPassword = ConfUtils.getString(conf, "http.proxy.pass", null);
+
+                // assemble proxy connection string
+                String proxyString = proxyType.toLowerCase() + "://";
+
+                // conditionally append authentication info
+                if (proxyUsername != null && !proxyUsername.isEmpty() &&
+                        proxyPassword != null && !proxyPassword.isEmpty()) {
+                    proxyString += proxyUsername + ":" + proxyPassword + "@";
+                }
+
+                // complete proxy string
+                proxyString += String.format("%s:%d", proxyHost, proxyPort);
+
+                try {
+                    // create single proxy manager
+                    proxyManager = (ProxyManager) proxyManagerClass.newInstance();
+                    proxyManager.configure(proxyRotationScheme, proxyString);
+                } catch (FileNotFoundException | InstantiationException | IllegalAccessException e) {
+                    LOG.error("failed to create proxy manager `" + proxyString + "`", e);
+                    success = false;
+                }
+            }
+
+            // get current time to track wait timeout
+            long start = System.currentTimeMillis();
+
+            // wait until proxy manager is ready with a 2 minute timeout
+            while (!proxyManager.ready()) {
```

Comment:
configure the timeout and sleep values instead of hardcoding; put the default key vals in the default config file

Reply:
In addition, I think it would be best to handle the wait time within the class constructor. This made more sense for my specific application then for the community as a whole